server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -360,7 +360,9 @@ public short partitionRecordVersion() {
     }
 
     public short fetchRequestVersion() {
-        if (this.isAtLeast(IBP_3_5_IV1)) {
+        if (this.isAtLeast(IBP_3_7_IV0)) {
```

Comment:
Does this work because all the new fields are tagged (so no real changes in handling)?

Reply:
This is supposed to be the first approach mentioned in this comment https://github.com/apache/kafka/pull/14444#discussion_r1360994577

My understanding is clusters would first be upgraded to 3.7 and then the IBP would be bumped after all the brokers are upgraded, but please correct me if that's wrong. You're right though, all the fields are tagged and there's no change in handling on the broker side.

Reply:
I was referring to whether the code here that we want to merge will break the build if we haven't implemented the new fields.

Reply:
Oh, yeah we shouldn't need any other changes, the fields are all tagged and not getting used anywhere so we can leave them as their default values.

Reply:
I just realized that if we set the version as unstable, we may not be able to use it here. 🤦‍♀️ Maybe we should remove the unstable version true if this causes issues in tests.

Sorry for confusion.